README.md

```diff
@@ -0,0 +1,117 @@
+# Strata-Rust
+
+Strata-Rust is a unique way of exploring Rust code, aiming to answer
+questions like:
+
+- Which methods take a generic argument that implements the trait `Foo`?
+- Do any structs contain a field called `data`?
+- What statements come after a "TODO" comment?
+
+## How do I use it?
+
+Right now, this project is very much in an alpha state. That's the
+polite way of saying "it works on my machine, and probably no
+other". I'm hoping to get it into a shape where it at least compiles
+in CI soon.
+
+## Components
+
+Strata-Rust is a combination of many pieces of open source
+technology. Some selected pieces include:
+
+### Strata
+
+[Strata][] is the underlying search technology that powers the ability
+to perform queries based on the structure of the data. These queries
+are different from the traditional search engine inverted index and
+are based around manipulating ranges of the indexed text known as
+"extents".
+
+The Strata crate works well on stable Rust.
+
+[Strata]: https://github.com/shepmaster/strata
+
+### Untitled Rust Parser
+
+While there are many high-quality Rust parsers available, a slightly
+different kind of parser was needed to provide the detail needed for
+the extents. To that end, a new parser was built using the [Peresil][]
+parsing toolkit. The parser aims to:
+
+- be a low-level parser of Rust source code, providing unshackled
+  access to the direct parsing structures.
+- provide an easier-to-use visitor interface that allows quickly
+  gathering information about Rust code.
+- provide reasonable quality errors about the parsing.
+- parse all syntactically-valid Rust code.
+
+It has anti-goals as well! The parser does not:
+
+- attempt to perform *semantic* analysis of Rust code.
+- guarantee to reject all syntactically-invalid code.
+
+It is expected that this parser will eventually grow a name and be
+extracted from this project. Currently it is of alpha quality and we
+know there is valid syntax that cannot be parsed.
+
+While Peresil works with stable Rust, our usage of it enables some
+**nightly-only** features.
```

Comment:
> When I looked, none of the nightly only features seemed essential to the implementation of the library

- `field_init_shorthand` - easily transformable, but my goodness does typing `{foo: foo, bar: bar}` get tiresome fast
- `pattern` - mostly out of laziness
- `custom_derive` - vestigial, only because I started before Rust 1.15

> the only one which seemed to have an important semantic impact being `conservative_impl_trait`

Yep this is the big one. Without this, parsing becomes a lot uglier and/or inefficient. I haven't done the transformation because it will be annoying, but theoretically all of the `impl Foo` can become `Box<Foo>`. The sad thing is that there's no trick to reuse one implementation for the other; they'd have to be completely copy-and-pasted.

> Do you intend to use more nightly features?

There are none that I'm currently aware of that would be applicable. If the whole enum-variant-as-separate type ever comes about, I'd use that in a heartbeat. That would reduce a lot of clutter of the data types.